tests/storage-vm

```diff
@@ -281,6 +281,12 @@ for poolDriver in $poolDriverList; do
         echo "==> Testing VM optimized export/import (while running to check config.mount is excluded)"
         ! lxc export v1 --quiet --compression=none - --optimized-storage | tar -tf - | grep -F config.mount || false
 
+        if [[ $(hasNeededAPIExtension vm_zfs_storage_disk_live_resize) && $(hasNeededAPIExtension vm_lvm_storage_disk_live_resize) && ( "${poolDriver}" == "zfs" || "${poolDriver}" == "lvm" ) ]]; then
+                echo "==> Checking live root disk device resize support with ${poolDriver} backend"
+                lxc config device set v1 root size=16GiB
+                [ "$(($(lxc exec v1 -- blockdev --getsize64 /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_root) / GiB))" -eq "16" ]
+        fi
+
         echo "==> Increasing VM root disk size for next boot (24GiB)"
         lxc config device set v1 root size=24GiB
         [ "$(lxc config get v1 volatile.root.apply_quota)" = "true" ]
```

Comment:
I suspect this ^ will fail on the live-growable backends as the growing won't be deferred anymore.

Reply:
This should pass with new change - here's a copy of my local test with a build of https://github.com/canonical/lxd/pull/14211. 

```bash
❯ lxc launch ubuntu:n --vm v1 --storage=zfs
Launching v1
                                                   
❯ lxc config device set v1 root size=16GiB

❯ lxc exec v1 -- blockdev --getsize64 /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_root | awk '{printf "%.2f GiB\n", $1 / (1024^3)}'
```